File: leo2-central/src/test/java/org/deku/leo2/central/data/DataTest.java
package org.deku.leo2.central.data;

import org.deku.leo2.central.PersistenceContext;
import org.junit.runner.RunWith;
import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.context.support.AnnotationConfigContextLoader;

/**
 * Created by masc on 15.05.15.
 */
@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = {
        PersistenceContext.Central.class,
        DataTest.TestConfiguration.class}, loader = AnnotationConfigContextLoader.class)
public class DataTest implements ApplicationContextAware {
    @Configuration
    @ComponentScan(basePackages = {"org.deku.leo2.central"})
    static class TestConfiguration {}

    @Override
    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {

    }
}

File: leoz-node/src/test/java/org/deku/leoz/node/test/DataTest.kt
package org.deku.leoz.node.test

import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.conf.global
import org.deku.leoz.node.Application
import org.deku.leoz.node.Storage
import org.deku.leoz.node.config.*
import org.junit.Ignore
import org.junit.runner.RunWith
import org.springframework.boot.autoconfigure.flyway.FlywayAutoConfiguration
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner

/**
 * Created by masc on 15.05.15.
 */
@Ignore
@RunWith(SpringJUnit4ClassRunner::class)
@SpringBootTest(classes = arrayOf(
        ApplicationConfiguration::class,
        PersistenceConfiguration::class,
        FlywayAutoConfiguration::class,
        ExecutorConfiguration::class
))
@EnableConfigurationProperties
@ActiveProfiles(Application.PROFILE_CLIENT_NODE)
open class DataTest {
    companion object {
        init {
            Kodein.global.addImport(ApplicationConfiguration.module)
            Kodein.global.addImport(StorageConfiguration.module)
            Kodein.global.addImport(LogConfiguration.module)
        }
    }
}
